### PR TITLE
Version 0.1

### PR DESCRIPTION
**Pull request by XedinUnknown**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,4 +8,54 @@
 
 Common abstract functionality for output renderers.
 
-[Dhii]: https://github.com/Dhii/dhii
+## Details
+This package provides abstract functionality for the most common implementations
+of output renderers. The classes are meant to implement functionality for
+interfaces in [`dhii/output-renderer-interface`], although of course they can
+be used for other purposes. This is a good place to start of you are creating
```
Comment (mecha) [inline README.md]: Typo: _"This is a good place to start ~of~ if"_

**Review by mecha — CHANGES_REQUESTED** [review]

**Review by mecha — APPROVED** [review]

File: test/unit/AbstractTemplateTest.php [hunk 2]
```diff
@@ -38,15 +38,65 @@ public function createInstance()
...
+    {
+        $paddingClassName = uniqid($className);
+        $definition = vsprintf('abstract class %1$s extends %2$s implements %3$s {}', [
+            $paddingClassName,
+            $className,
+            implode(', ', $interfaceNames),
+        ]);
+        eval($definition);
+
+        return $this->getMockForAbstractClass($paddingClassName);
```
Comment (mecha) [inline test/unit/AbstractTemplateTest.php:64]: I recommend that you return `$this->getMockBuilder($paddingClassName)` instead. See other comment.

File: test/unit/AbstractTemplateTest.php [hunk 3]
```diff
@@ -38,15 +38,65 @@ public function createInstance()
...
      *
-     * @return ValidationFailedException
+     * @return ValidationFailedExceptionInterface
      */
     public function createValidationFailedException()
     {
-        $mock = $this->getMock('Dhii\Validation\Exception\ValidationFailedException');
+        $mock = $this->mockClassAndInterfaces('Exception', ['Dhii\Validation\Exception\ValidationFailedExceptionInterface']);
+        $mock->method('getValidationErrors')
+                ->will($this->returnValue(null));
```
Comment (mecha) [inline test/unit/AbstractTemplateTest.php]: There is no need to do this for every method. If you use a mock builder, you can do

```
$mockBuilder->setMethods(
    [
        'getValidationErrors',
        'getValidator',
        ...
    ]
)
```

and they will all be mocked to return `null` by default. In the end, you can then

```
return $mockBuilder->getMockForAbstractClass();
```

**Review by mecha — COMMENTED** [review]
Just a quick rant on code cleanliness 😛

**Review by mecha — APPROVED** [review]
LGTM. If you're interested in some code cleanup duty, see my previous review. But it's otherwise unimportant.

**Comment by XedinUnknown** [comment]
It appears there's more work to be done before initial release can happen. See #3.

**Pull request merged**